### PR TITLE
chore(storage-browser): add Message feedback to LocationDetailView

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

- Outputs an error <Message /> in LocationDetailView if hasError is true.

Tested by loading the app, setting network to offline, the clicking on a location.

**screenshot**

Screenshot 2024-08-19 at 2 44 47 PM

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by timngyn — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts [hunk 1]
```diff
@@ -95,12 +95,16 @@ export async function listLocationItemsAction(
...
+    const result = [
+      ...(refresh ? [] : prevState.result),
+      ...parseResult(output, path),
+    ];
 
-  return { result, nextToken: output.nextToken };
+    return { result, nextToken: output.nextToken };
+  } catch (e) {
+    return { result: [], nextToken: undefined, message: (e as Error).message };
+  }
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts]: Error `message` is already captured and can be read from:
```ts
const [{ message }] = useAction({ type: 'LIST_LOCATION_ITEMS; });
```

File: packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts [hunk 2]
```diff
@@ -95,12 +95,20 @@ export async function listLocationItemsAction(
...
+    ];
+
+    return { result, nextToken: output.nextToken };
+  } catch (e) {
+    return {
+      result: [],
+      nextToken: undefined,
+      message: (e as Error).message || 'There was an error listing items.',
+    };
+  }
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts:113]: See https://github.com/aws-amplify/amplify-ui/pull/5609#discussion_r1722236897

File: packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts (hunk 1 above)
Comment (hbuchel) [reply packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts]: Thanks, updated in latest

**Review by timngyn — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**